One shot install script for CentOS 7

TheFiZi opened the pull request:
Ran a few test runs and everything seems good. Not 100% sure on the ports that need to be opened.

Sorry for the second pull request. Forgot to shill at the end of the script :)

TheFiZi committed:
One shot install script for CentOS 7

Ran a few test runs and everything seems good. Not 100% sure on the ports that need to be opened.

Sorry for the second pull request. Forgot to shill at the end of the script :)

TheFiZi committed:
Added yum-plugin-versionlock

Installed yum-plugin-versionlock and configured it to lock the golang version so it stays locked at v1.9.3 since none of this works with v1.9.4 or 1.10.x.

TheFiZi committed:
Bumped version number

Forgot to do so in my previous commit

automyr commented on hk_install_centos7.sh:
```diff
@@ -0,0 +1,289 @@
+echo "Port 14265 - TCP and UDP"
+echo "Port 14600 - UDP"
+echo "Port 15600 - TCP"
+echo ""
```
It's not crucial, but you should probably add port 18600 (TCP) for Nelson. You can add a comment to open the ports on the router if they want to check on their node from outside. I think you don't need to open the ports in the router if you're just calling localhost from inside.

automyr replied:
Actually, close down 14265 - it's unsafe to leave it exposed and the Go server will interact with API using localhost.

automyr commented on hk_install_centos7.sh:
```diff
@@ -0,0 +1,289 @@
+su - hooknode -c "mkdir -p /home/hooknode/go/bin"
+
+# Install the go dependancy manager
+su - hooknode -c "go get -u github.com/golang/dep/cmd/dep"
```
dep is already installed with the rest of the dependencies when you call `make install-deps`. It certainly doesn't hurt to have it before that, but I think it's superfluous.

TheFiZi replied:
Dep is not automatically installed on CentOS when you install golang. I had to manually install it to get things working otherwise `make install-deps` would fail.

TheFiZi committed:
Changed firewall port recommendations

Removed opening 14265 (TCP/UDP) on firewalld and recommendation to forward port

Added opening 18600 (TCP) for access to Nelson to view node status along with additional port forwarding instructions at the end of the script.

automyr commented on hk_install_centos7.sh:
```diff
@@ -271,13 +271,16 @@ internalIps="$(echo -e "${internalIps}" | tr -d '[:space:]')"
-echo "Port 14265 - TCP and UDP"
 echo "Port 14600 - UDP"
 echo "Port 15600 - TCP"
+echo "Port 15600 - TCP - Optional if you want to view your node status remotely"
```
This should actually be 18600, not 15600 :)

TheFiZi committed:
Fixed typo

Should be 18600 and not 15600
